### ISSUE TITLE
Crate Licensing

### ISSUE THREAD

**Issue by mattyg** [holochain/holochain#2887]
Looks like a bunch of crates are licensed under MIT or Apache 2.0 according to the their README and Cargo.toml. Should they be licensed under CAL?

**Comment by ThetaSinner** [comment]
Yeah the licensing looks a bit weird, I queried it too but there is a deliberate difference. I will do my best to explain 😄 

The CAL license is there to protect `holochain` itself, not all the library code which anybody who wants to can build something new with. So to quote @artbrock 

> CAL is for Holochain. It is a the first license designed for P2P use. The important difference is that if you use this software under CAL then you are required to allow peers to control their own private keys and data.

Though there is probably more information about the CAL elsewhere.

So really only `holochain` should be under the CAL. Everything else should be either Apache 2 or dual licensed under Apache 2 + MIT is okay too. I see we do have a few libraries that are licensed under CAL when they don't need to be.

As an example, if you built something new with Kitsune then your final binary would not be required to comply with the CAL. Unless you chose to license under it that is.

I do think there is some cleanup to be done here so we'll use this issue for that!

**Comment by artbrock** [comment]
I would agree that general purpose libraries could be Apache 2 or MIT (not sure why to dual license them). The notion was Apache was better for some corporate IP compatibility -- I forget the exact details. MIT would be for little utilities we don't care about any corporate IP for.

However, CAL may apply to more than just Holochain. You used kitsune as an example, and it revolves around P2P cryptographic keys... shouldn't those have the CAL end-user right guarantees?

**Comment by ThetaSinner** [comment]
I'd say it's a choice. Kitsune requires its host environment to manage keys on its behalf so it's up to the behaviour of the binary that uses Kitsune to decide how that works. It makes sense to me that we apply the CAL license to it, as it is more or less independent of Holochain and is largely re-usable.

I've updated the PR with `kitsune-p2p` switched to the CAL license

**Comment by jost-s** [comment]
What about the Holochain clients? Currently they're under CAL and this sounds like that's appropriate.

**Comment by ThetaSinner** [comment]
@jost-s What was the outcome of the discussion about the clients? and am I okay to close this issue?

**Comment by jost-s** [comment]
JS client is going to be under CAL for the time being, Rust client and Tryorama under MIT.

**Comment by ThetaSinner** [comment]
Great! I suppose we should check wasmer, hsb and tx5 while we're at it

**Comment by ThetaSinner** [comment]
> Great! I suppose we should check wasmer, hsb and tx5 while we're at it

And now SBD too, we should check these
